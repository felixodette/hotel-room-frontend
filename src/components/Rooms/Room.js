import React, { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';

const Room = () => {
  const [roomDetails, setRoomDetails] = useState('');
  const { id } = useParams();

  useEffect(() => {
    const roomData = {
      method: 'GET',
      headers: { 'Content-Type': 'application/json' },
    };
    fetch(`http://localhost:3001/api/v1/rooms/${id}`, roomData)
      .then((response) => response.json())
      .then((data) => setRoomDetails(data));
  }, []);

  return (
    <div>
      <button type="button">Reserve</button>
      <h1>
        {' '}
        Room name:
        {roomDetails.name}
        {' '}
      </h1>
      <h1>
        {' '}
        Room description:
        {roomDetails.description}
        {' '}
      </h1>
      <h1>
        <img src={roomDetails.image} alt={roomDetails.name} />
      </h1>
      <h1>
        {' '}
        Room size:
        {roomDetails.size}
        {' '}
      </h1>
      <h1>
        {' '}
        Room view:
        {roomDetails.view}
        {' '}
      </h1>
      <h1>
        {' '}
        Room bedding:
        {roomDetails.bedding}
        {' '}
      </h1>
    </div>
  );
};
export default Room;
