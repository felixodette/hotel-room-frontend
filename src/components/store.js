import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from '../redux/reducers/index';

const initialState = {};
const middleware = [thunk];
const store = createStore(
  rootReducer,
  initialState,
  compose(
    applyMiddleware(...middleware),
/* eslint-disable */
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()));
/* eslint-enable */

export default store;
