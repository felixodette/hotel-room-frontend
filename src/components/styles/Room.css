.room-div {
  padding: 5rem 0 0 1.5rem;
}

.room-image {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.room-details {
  padding: 1rem;
}

.room-name {
  text-align: end;
  padding: 0 2rem 1rem 0;
}

.room-data :nth-child(2n+0) {
  background-color: transparent;
}

.room-data p {
  display: flex;
  justify-content: space-between;
  background-color: rgb(232, 229, 229);
  padding: 0.5rem 2rem 0.5rem 0;
}

.reserve-btn {
  padding: 25% 0 0 0;
  display: flex;
  justify-content: center;
}

.reserve-btn button {
  background-color: #97bf11;
  color: #fff;
  border-radius: 20px;
  font-size: 24px;
  width: 12rem;
  display: flex;
  justify-content: space-between;
  height: 40px;
  transition-duration: 0.4s;
  padding: 0 7px;
  border: none;
}

.back-arrow {
  padding: 0 0 3rem 0;
  margin-left: 13rem;
}

.back-arrow button {
  border-radius: 0 50% 50% 0;
  width: 80px;
  height: 60px;
  font-size: 24px;
  color: #fff;
  font-weight: bold;
  box-shadow: 0 0 10px 0 rgba(0, 0, 0, 0.2);
  background-color: #97bf11;
}

.back-arrow button:hover {
  background-color: #97bf11;
  color: #fff;
}
