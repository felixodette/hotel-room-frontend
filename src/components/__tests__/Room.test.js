import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import Room from '../Rooms/Room';
import store from '../../redux/configureStore';

describe('A Room', () => {
  render(
    <Provider store={store}>
      <Router>
        <Room />
      </Router>
    </Provider>,
  );

  test('Test the dom for successful rendering of elements', () => {
    const RoomDescription = screen.getByText('Room description:');
    expect(RoomDescription).toBeInTheDocument();
  });
});
