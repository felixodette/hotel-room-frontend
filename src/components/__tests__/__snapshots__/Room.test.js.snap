// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A Room It correctly renders a room component 1`] = `
<div>
  <div
    class="row room-div"
  >
    <div
      class="col-md-8 room-image"
    >
      <img
        alt="room"
        class="room-image"
      />
    </div>
    <div
      class="col-md-4 room-details"
    >
      <h3
        class="room-name"
      />
      <div
        class="room-data"
      >
        <p
          class="room-description"
        >
          Room description:
          <span />
        </p>
        <p
          class="room-size"
        >
          Room size:
          <span />
        </p>
        <p
          class="room-view"
        >
          Room view:
          <span />
        </p>
        <p
          class="room-bedding"
        >
          Room bedding:
          <span />
        </p>
        <div
          class="reserve-btn"
        >
          <button
            class="fw-bold"
            type="button"
          >
            Reserve
            <span>
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M210.7 147.6c7.5-7.5 19.8-7.5 27.3 0l95.4 95.7c7.3 7.3 7.5 19.1.6 26.6l-94 94.3c-3.8 3.8-8.7 5.7-13.7 5.7-4.9 0-9.9-1.9-13.6-5.6-7.5-7.5-7.6-19.7 0-27.3l79.9-81.1-81.9-81.1c-7.6-7.4-7.6-19.6 0-27.2z"
                />
                <path
                  d="M48 256c0 114.9 93.1 208 208 208s208-93.1 208-208S370.9 48 256 48 48 141.1 48 256zm32 0c0-47 18.3-91.2 51.6-124.4C164.8 98.3 209 80 256 80s91.2 18.3 124.4 51.6C413.7 164.8 432 209 432 256s-18.3 91.2-51.6 124.4C347.2 413.7 303 432 256 432s-91.2-18.3-124.4-51.6C98.3 347.2 80 303 80 256z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="back-arrow fixed-bottom"
    >
      <button
        class="btn"
        type="button"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18.464 2.114a.998.998 0 0 0-1.033.063l-13 9a1.003 1.003 0 0 0 0 1.645l13 9A1 1 0 0 0 19 21V3a1 1 0 0 0-.536-.886zM17 19.091 6.757 12 17 4.909v14.182z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;

exports[`A Room It renders the Room correctly 1`] = `
<div
  className="row room-div"
>
  <div
    className="col-md-8 room-image"
  >
    <img
      alt="room"
      className="room-image"
    />
  </div>
  <div
    className="col-md-4 room-details"
  >
    <h3
      className="room-name"
    />
    <div
      className="room-data"
    >
      <p
        className="room-description"
      >
        Room description:
        <span />
      </p>
      <p
        className="room-size"
      >
        Room size:
        <span />
      </p>
      <p
        className="room-view"
      >
        Room view:
        <span />
      </p>
      <p
        className="room-bedding"
      >
        Room bedding:
        <span />
      </p>
      <div
        className="reserve-btn"
      >
        <button
          className="fw-bold"
          onClick={[Function]}
          type="button"
        >
          Reserve
          <span>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M210.7 147.6c7.5-7.5 19.8-7.5 27.3 0l95.4 95.7c7.3 7.3 7.5 19.1.6 26.6l-94 94.3c-3.8 3.8-8.7 5.7-13.7 5.7-4.9 0-9.9-1.9-13.6-5.6-7.5-7.5-7.6-19.7 0-27.3l79.9-81.1-81.9-81.1c-7.6-7.4-7.6-19.6 0-27.2z"
              />
              <path
                d="M48 256c0 114.9 93.1 208 208 208s208-93.1 208-208S370.9 48 256 48 48 141.1 48 256zm32 0c0-47 18.3-91.2 51.6-124.4C164.8 98.3 209 80 256 80s91.2 18.3 124.4 51.6C413.7 164.8 432 209 432 256s-18.3 91.2-51.6 124.4C347.2 413.7 303 432 256 432s-91.2-18.3-124.4-51.6C98.3 347.2 80 303 80 256z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    className="back-arrow fixed-bottom"
  >
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M18.464 2.114a.998.998 0 0 0-1.033.063l-13 9a1.003 1.003 0 0 0 0 1.645l13 9A1 1 0 0 0 19 21V3a1 1 0 0 0-.536-.886zM17 19.091 6.757 12 17 4.909v14.182z"
        />
      </svg>
    </button>
  </div>
</div>
`;
